ipinfo/config.go

```diff
@@ -32,74 +32,127 @@ func getConfigDir() (string, error) {
 	return confDir, nil
 }
 
-// init function
+// returns the path to the config file.
+func ConfigPath() (string, error) {
+	confDir, err := getConfigDir()
+	if err != nil {
+		return "", err
+	}
+
+	return filepath.Join(confDir, "config.json"), nil
+}
+
+// returns the path to the token file.
+func TokenPath() (string, error) {
+	confDir, err := getConfigDir()
+	if err != nil {
+		return "", err
+	}
+
+	return filepath.Join(confDir, "token"), nil
+}
+
 func InitConfig() error {
-	path, err := ConfigPath()
+	configpath, err := ConfigPath()
+	if err != nil {
+		return err
+	}
+
+	tokenpath, err := TokenPath()
 	if err != nil {
 		return err
 	}
 
-	if !lib.FileExists(path) {
-		err := SetConfig(NewConfig())
+	// create default config if none yet.
+	if !lib.FileExists(configpath) {
+		gConfig = NewConfig()
+	} else {
+		config, err := ReadConfig()
 		if err != nil {
 			return err
 		}
+		gConfig = config
 	}
-	return nil
-}
 
-// returns the path to config file.
-func ConfigPath() (string, error) {
-	confDir, err := getConfigDir()
-	if err != nil {
-		return "", err
+	// if token exists, migrate it to config file.
+	if lib.FileExists(tokenpath) {
+		token, err := TokentoConfig()
+		if err != nil {
+			return err
+		}
+		gConfig.Token = token
+
+		// remove the existing token file
+		if err := os.Remove(tokenpath); err != nil {
+			return err
+		}
 	}
-	return filepath.Join(confDir, "config.json"), nil
+	if err := SaveConfig(gConfig); err != nil {
+		return err
+	}
+
+	return nil
 }
 
-// returns config file with default settings.
+// returns a new, default config.
 func NewConfig() Config {
 	return Config{
-		Cache: true,
-		Token: "",
+		CacheEnabled: true,
+		Token:        "",
 	}
 }
 
-// set the values of config.
-func SetConfig(config Config) error {
+// migration of token to config file.
+//
+// might be deleted in future release.
+func TokentoConfig() (string, error) {
```

Comment:
The actual logic here is to read the token value from the old token file, so it would be better renamed to `ReadTokenFile`. The documentation should also be updated.